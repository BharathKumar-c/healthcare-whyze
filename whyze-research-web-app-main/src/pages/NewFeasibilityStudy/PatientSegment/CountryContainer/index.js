import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'antd';
import { Star, StarFilled, Tick } from '../../../../assets';
import { Text } from '../../../../components/basic';
import { totalPatientFormatter } from '../../../../utils/calculations';
import { PlusOutlined } from '@ant-design/icons';

export default function CountryContainer({
  isPinned,
  countryObj,
  onUnPinCountry,
  onPinCountry,
}) {
  const { country, count } = countryObj;
  return (
    <div className="country-container">
      <Row>
        <Col span={18}>
          <Row className="">
            <Col
              className="country-container-column"
              span={4}
              style={{ cursor: 'pointer' }}
            >
              {isPinned ? (
                <StarFilled
                  onClick={() => onUnPinCountry(country)}
                  className="favorite-isFilled"
                />
              ) : (
                <Star
                  onClick={() => onPinCountry(country)}
                  className="favorite-notFilled"
                />
              )}
            </Col>
            <Col
              span={20}
              style={{
                whiteSpace: 'nowrap',
                overflow: 'hidden',
                textOverflow: 'ellipsis',
              }}
            >
              <Text
                className={
                  isPinned
                    ? 'patient-segment-pinned-title'
                    : 'patient-segment-title'
                }
                text={country ? country : 'Country not found'}
              />
            </Col>
          </Row>
        </Col>
        <Col span={6}>
          <Row style={{ alignItems: 'flex-end' }}>
            <Col
              span={14}
              className={
                isPinned ? 'no-of-patients' : 'country-container-no-of-patients'
              }
            >
              {totalPatientFormatter(count)}
            </Col>
            <Col span={10} style={{ cursor: 'pointer' }}>
              {isPinned ? (
                <Tick
                  onClick={() => console.log('tick', country)}
                  className="Tick-icon"
                />
              ) : (
                <PlusOutlined
                  onClick={() => console.log('add', country)}
                  className="Pluse-Icon"
                />
              )}
            </Col>
          </Row>
        </Col>
      </Row>
    </div>
  );
}

CountryContainer.propTypes = {
  isPinned: PropTypes.bool,
  countryObj: PropTypes.objectOf(
    PropTypes.shape({
      country: PropTypes.object,
      count: PropTypes.number,
    }),
  ),
  onUnPinCountry: PropTypes.func,
  onPinCountry: PropTypes.func,
};

CountryContainer.defaultProps = {
  isPinned: false,
  countryObj: {
    country: '',
    count: 0,
  },
  onUnPinCountry: () => {},
  onPinCountry: () => {},
};
