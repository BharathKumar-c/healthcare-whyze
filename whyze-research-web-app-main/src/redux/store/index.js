import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
