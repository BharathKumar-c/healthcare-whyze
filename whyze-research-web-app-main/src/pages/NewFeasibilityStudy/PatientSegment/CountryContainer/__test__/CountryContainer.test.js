import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import CountryContainer from '..';
import React from 'react';

const democountryList = { Country: 'United Kingdom', count: 1 };

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
describe('test cases for CountryContainer', () => {
  test('create snapshots', () => {
    const { container } = renderWithContext(<CountryContainer />);
    expect(container).toMatchSnapshot();
  });
  test('should test unPinned country ', () => {
    const { container } = renderWithContext(
      <CountryContainer isPinned={false} country={democountryList} />,
    );
    const favoriteFilled =
      container.getElementsByClassName('favorite-notFilled');
    fireEvent.click(favoriteFilled[0].firstChild);
    expect(container.getElementsByClassName('country-container').length).toBe(
      1,
    );
  });
  test('should test isPinned country', () => {
    const { container } = renderWithContext(
      <CountryContainer isPinned={true} country={democountryList} />,
    );
    const favoriteFilled =
      container.getElementsByClassName('favorite-isFilled');
    fireEvent.click(favoriteFilled[0].firstChild);
    expect(container.getElementsByClassName('country-container').length).toBe(
      1,
    );
  });
  test('should test tickIconClicked country', () => {
    const { container } = renderWithContext(
      <CountryContainer isPinned={true} country={democountryList} />,
    );
    const tickIconClicked = container.getElementsByClassName('Tick-icon');
    fireEvent.click(tickIconClicked[0].firstChild);
    expect(tickIconClicked[0].firstChild.textContent).toBe('Tick.svg');
  });
  test('should test pluseIconClicked  country', () => {
    const { container } = renderWithContext(
      <CountryContainer isPinned={false} country={democountryList} />,
    );
    const pluseIconClicked = container.getElementsByClassName('Pluse-Icon');
    fireEvent.click(pluseIconClicked[0].firstChild);
  });
});
