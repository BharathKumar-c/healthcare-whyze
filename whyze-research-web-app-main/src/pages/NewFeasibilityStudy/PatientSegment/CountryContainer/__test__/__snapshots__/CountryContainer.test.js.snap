// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for CountryContainer create snapshots 1`] = `
<div>
  <div
    class="country-container"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-18"
      >
        <div
          class="ant-row"
        >
          <div
            class="ant-col ant-col-4"
            style="cursor: pointer;"
          >
            <svg
              class="favorite-notFilled"
            >
              Star.svg
            </svg>
          </div>
          <div
            class="ant-col ant-col-20"
            style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
          >
            <span
              class="ant-typography patient-segment-title"
            >
              Country not found
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-6"
      >
        <div
          class="ant-row"
          style="align-items: flex-end;"
        >
          <div
            class="ant-col ant-col-14 country-container-no-of-patients"
          >
            0
          </div>
          <div
            class="ant-col ant-col-10"
            style="cursor: pointer;"
          >
            <span
              aria-label="plus"
              class="anticon anticon-plus Pluse-Icon"
              role="img"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                data-icon="plus"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                />
                <path
                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
